softgroup/basis/common-dao/src/main/java/com/softgroup/common/dao/api/AbstractCrudService.java

```diff
@@ -0,0 +1,42 @@
+package com.softgroup.common.dao.api;
+
+import com.softgroup.common.dao.repository.ProfileRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Created by yuriy on 21.03.17.
+ */
+public class AbstractCrudService<T, ID extends Serializable, R extends PagingAndSortingRepository<T, ID>> {
```

Comment:
@Yuriy27 Our entities will use String IDs. Make some common base repository with String ID - it will simplify generics.